elasticsearch.go

```diff
@@ -175,99 +171,81 @@ func (es *Elasticsearch) CreateUser(ctx context.Context, statements dbplugin.Sta
-		return errwrap.Wrapf("unable to get client: {{err}}", err)
+		return newdbplugin.DeleteUserResponse{}, errwrap.Wrapf("unable to get client: {{err}}", err)
 	}
 
 	var errs error
-	if len(stmt.RoleToCreate) > 0 {
-		// If the role already doesn't exist because it was successfully deleted on a previous
-		// attempt to run this code, there will be no error, so it's harmless to try.
-		if err := client.DeleteRole(ctx, username); err != nil {
-			errs = multierror.Append(errs, errwrap.Wrapf(fmt.Sprintf("unable to delete role name %s: {{err}}", username), err))
-		}
+	// If the role already doesn't exist, either it wasn't created for this
+	// user, or it was successfully deleted on a previous attempt to run this
+	// code, there will be no error, so it's harmless to try.
+	if err := client.DeleteRole(ctx, req.Username); err != nil {
+		errs = multierror.Append(errs, errwrap.Wrapf(fmt.Sprintf("unable to delete role name %s: {{err}}", req.Username), err))
 	}
+
 	// Same with the user. If it was already deleted on a previous attempt, there won't be an
 	// error.
-	if err := client.DeleteUser(ctx, username); err != nil {
-		errs = multierror.Append(errs, errwrap.Wrapf(fmt.Sprintf("unable to create user name %s: {{err}}", username), err))
+	if err := client.DeleteUser(ctx, req.Username); err != nil {
+		errs = multierror.Append(errs, errwrap.Wrapf(fmt.Sprintf("unable to delete user name %s: {{err}}", req.Username), err))
 	}
-	return errs
+	return newdbplugin.DeleteUserResponse{}, errs
 }
 
-// RotateRootCredentials doesn't require any statements from the user because it's not configurable in any
+// UpdateUser doesn't require any statements from the user because it's not configurable in any
 // way. We simply generate a new password and hit a pre-defined Elasticsearch REST API to rotate them.
-func (es *Elasticsearch) RotateRootCredentials(ctx context.Context, _ []string) (map[string]interface{}, error) {
-	newPassword, err := es.credentialProducer.GeneratePassword()
-	if err != nil {
-		return nil, errwrap.Wrapf("unable to generate root password: {{err}}", err)
-	}
+func (es *Elasticsearch) UpdateUser(ctx context.Context, req newdbplugin.UpdateUserRequest) (newdbplugin.UpdateUserResponse, error) {
 
 	// Don't let anyone read or write the config while we're in the process of rotating the password.
 	es.mux.Lock()
 	defer es.mux.Unlock()
 
 	client, err := buildClient(es.config)
 	if err != nil {
-		return nil, errwrap.Wrapf("unable to get client: {{err}}", err)
+		return newdbplugin.UpdateUserResponse{}, fmt.Errorf("unable to get client: %w", err)
 	}
 
-	if err := client.ChangePassword(ctx, es.config["username"].(string), newPassword); err != nil {
-		return nil, errwrap.Wrapf("unable to change password: {{}}", err)
+	if req.Password != nil {
+		if err := client.ChangePassword(ctx, req.Username, req.Password.NewPassword); err != nil {
+			return newdbplugin.UpdateUserResponse{}, fmt.Errorf("unable to change password: %w", err)
+		}
+
```

Comment:
nit: can you add a comment explicitly saying the expiration change is a no-op?

Reply:
added in 507cc5319bdb4adbd706ad866cd591d087de9dfc